contracts/apps/voting/VotingApp.sol

```diff
@@ -0,0 +1,190 @@
+
+        votings.length += 1;
+    }
+
+    /**
+    * @notice Change minimum acceptance quorum to `_minAcceptQuorumPct`
+    * @param _minAcceptQuorumPct New acceptance quorum
+    */
+    function changeMinAcceptQuorumPct(uint256 _minAcceptQuorumPct) auth external {
+        require(supportRequiredPct >= _minAcceptQuorumPct);
+        minAcceptQuorumPct = _minAcceptQuorumPct;
+    }
+
+    /**
+    * @notice Create new voting to execute `_executionScript`
+    * @param _executionScript EVM script to be executed on approval
+    * @return votingId id for newly created vote
+    */
+    function newVoting(bytes _executionScript) auth external returns (uint256 votingId) {
+        votingId = votings.length++;
+        Voting storage voting = votings[votingId];
+        voting.executionScript = _executionScript;
+        voting.creator = msg.sender;
+        voting.startDate = uint64(now);
+        voting.open = true;
+        voting.snapshotBlock = getBlockNumber() - 1; // avoid double voting in this very block
+        voting.totalVoters = token.totalSupplyAt(voting.snapshotBlock);
+
+        StartVote(votingId);
+
+        if (canVote(votingId, msg.sender)) _vote(votingId, true, msg.sender);
+    }
+
+    /**
+    * @notice Vote `_supports` in vote with id `_votingId`
+    * @param _votingId Id for vote
+    * @param _supports Whether voter supports the voting
+    */
+    function vote(uint256 _votingId, bool _supports) external {
+        require(canVote(_votingId, msg.sender));
+        _vote(_votingId, _supports, msg.sender);
+    }
+
+    /**
+    * @notice Execute the result of vote `_votingId`
+    * @param _votingId Id for vote
+    */
+    function executeVote(uint256 _votingId) external {
+        require(canExecute(_votingId));
+        _executeVote(_votingId);
+    }
+
+    function canVote(uint256 _votingId, address _voter) constant returns (bool) {
+        Voting storage voting = votings[_votingId];
+
+        return voting.open &&
+               uint64(now) < voting.startDate + voteTime &&
+               token.balanceOfAt(_voter, voting.snapshotBlock) > 0;
+    }
+
+    function canExecute(uint256 _votingId) constant returns (bool) {
+        Voting storage voting = votings[_votingId];
+
+        // Voting is already decided
+        if (voting.yea >= pct(voting.totalVoters, supportRequiredPct))
+            return true;
+
+        uint256 totalVotes = voting.yea + voting.nay;
+        if (uint64(now) >= voting.startDate + voteTime &&
+            voting.yea >= pct(totalVotes, supportRequiredPct) &&
+            voting.yea >= pct(voting.totalVoters, minAcceptQuorumPct))
+            return true;
+
+        return false;
+    }
+
+    function getVoting(uint256 _votingId) constant returns (bool open, bool executed, address creator, uint64 startDate, uint256 snapshotBlock, uint256 yea, uint256 nay, uint256 totalVoters, bytes32 scriptHash, uint256 scriptActionsCount) {
+        Voting storage voting = votings[_votingId];
```

Comment:
:D literally 10 return values

Reply:
jesus 😂

Reply:
We should probably increase it by default. AFAIK the cost is 1 gas per 32 bytes of memory expansion